exonum/src/lib.rs

```diff
@@ -18,6 +18,26 @@
 
 #![deny(missing_debug_implementations, missing_docs, unsafe_code, bare_trait_objects)]
 #![cfg_attr(feature = "long_benchmarks", feature(test))]
+#![cfg_attr(feature = "cargo-clippy", deny(clippy_pedantic))]
+#![cfg_attr(
+    feature = "cargo-clippy",
+    allow(
+          option_map_unwrap_or_else,
+          // Next cast.. lints don't give alternatives.
+          cast_possible_wrap, cast_possible_truncation, cast_sign_loss,
+          // The lint does not work properly with desugaring and macro.
+          used_underscore_binding,
```

Comment:
What king of desugaring? As for macros: I propose to suppress this lint locally (inside of the macro or inside of the modules with heavy macros usage).